src/utils.ts

```diff
@@ -219,6 +221,43 @@ ajv.addFormat('domain', {
   }
 });
 
+function getDomainType(domain: string): DomainType {
+  const isEns = domain.endsWith('.eth');
+
+  const tokens = domain.split('.');
+
+  if (tokens.length === 2 && !isEns) return 'other-tld';
+  else if (tokens.length > 2) return 'subdomain';
+  else if (isEns) return 'ens';
+  else throw new Error('Invalid domain');
```

Comment:
Returning `Uncaught (in promise) Error: Invalid domain` if i try `console.log(await snapshot.utils.getEnsOwner('abcd'));` maybe it should be `0x00` like before?